sycl/source/detail/scheduler/commands.cpp

```diff
@@ -691,6 +703,12 @@ bool Command::enqueue(EnqueueResultT &EnqueueResult, BlockingT Blocking) {
     // Consider the command is successfully enqueued if return code is
     // CL_SUCCESS
     MEnqueueStatus = EnqueueResultT::SyclEnqueueSuccess;
+    if (MLeafCounter == 0 && supportsPostEnqueueCleanup() &&
```

Comment:
Can we do in GraphProcessor::enqueue?

Reply:
See https://github.com/intel/llvm/pull/5070#discussion_r772351809